### PR TITLE
Update PollSettings to support exponential backoff

### PR DESCRIPTION
**Pull request by chrisdunelm**
If this looks OK, I'll add the backoff code in Polling.cs, and extra tests.

### CODE REVIEW

File: src/Google.Api.Gax/PollSettings.cs [hunk 1]
```diff
@@ -29,16 +29,43 @@ public sealed class PollSettings
...
+        public double DelayMultiplier { get; }
+
+        /// <summary>
+        /// The maximum delay to use. If the increasing delay due to the delay multiplier exceeds this,
+        /// this maximum is used instead.
+        /// </summary>
+        public TimeSpan MaxDelay { get; }
+
+        /// <summary>
+        /// Creates poll settings from the given expiration and delay.
```
Comment (jskeet) [inline src/Google.Api.Gax/PollSettings.cs]: Maybe make it clear that it's then a constant delay.

File: src/Google.Api.Gax/PollSettings.cs [hunk 2]
```diff
@@ -29,16 +29,43 @@ public sealed class PollSettings
...
         /// <param name="expiration">The expiration to use in order to know when to stop polling. Must not be null.</param>
         /// <param name="delay">The delay between RPC calls. Must be non-negative.</param>
-        public PollSettings(Expiration expiration, TimeSpan delay)
+        /// <param name="delayMultiplier">The multiplier to apply to the delay on each iteration; must be greater or equal to 1.0.</param>
+        /// <param name="maxDelay">The maximum delay to use.</param>
+        public PollSettings(Expiration expiration, TimeSpan delay, double delayMultiplier, TimeSpan maxDelay)
         {
             Expiration = GaxPreconditions.CheckNotNull(expiration, nameof(expiration));
             Delay = GaxPreconditions.CheckNonNegativeDelay(delay, nameof(delay));
+            if (delayMultiplier < 1.0 || double.IsNaN(delayMultiplier))
```
Comment (jskeet) [inline src/Google.Api.Gax/PollSettings.cs]: Does infinity count as NaN? I can't remember.

Comment (chrisdunelm) [inline src/Google.Api.Gax/PollSettings.cs]: No. But this validation is copied from [BackoffSettings.cs](https://github.com/googleapis/gax-dotnet/blob/master/src/Google.Api.Gax.Grpc/BackoffSettings.cs#L54). I thought it best to make them consistent.

**Comment by chrisdunelm** [comment]
I think the case of an LRO that has a very unpredictable runtime makes this reasonable.
If the RPC may take just a few seconds, or a few hours; then exponential backoff does make sense.

File: src/Google.Api.Gax/PollSettings.cs (hunk 2 above)
Comment (jskeet) [inline src/Google.Api.Gax/PollSettings.cs]: Ouch. We should probably fix it in both places :) (Perhaps add a common method somewhere?)

Comment (chrisdunelm) [inline src/Google.Api.Gax/PollSettings.cs]: Is it worth it? Allowing something like double.MaxValue, but disallowing +infinity seems a little unnecessary.

File: src/Google.Api.Gax/PollSettings.cs (hunk 1 above)
Comment (chrisdunelm) [inline src/Google.Api.Gax/PollSettings.cs]: Done

**Comment by chrisdunelm** [comment]
PTAL

File: src/Google.Api.Gax/PollSettings.cs (hunk 2 above)
Comment (jskeet) [inline src/Google.Api.Gax/PollSettings.cs]: Fair enough :)

File: src/Google.Api.Gax/PollSettings.cs [hunk 3]
```diff
@@ -29,16 +29,57 @@ public sealed class PollSettings
...
+            DelayMultiplier = delayMultiplier;
+            MaxDelay = GaxPreconditions.CheckNonNegativeDelay(maxDelay, nameof(maxDelay));
+        }
+
+        /// <summary>
+        /// Works out the next delay from the current one, based on the multiplier and maximum.
+        /// </summary>
+        /// <param name="currentDelay">The current delay.</param>
+        /// <returns>The next delay.</returns>
+        internal TimeSpan NextDelay(TimeSpan currentDelay)
```
Comment (jskeet) [inline src/Google.Api.Gax/PollSettings.cs]: Random alternative implementation: return an `IEnumerable<TimeSpan>`. Not really suggesting it, but would be interesting.

File: src/Google.Api.Gax/Polling.cs [hunk 4]
```diff
@@ -59,15 +60,16 @@ public static class Polling
...
+                currentDelay = pollSettings.NextDelay(currentDelay);
             }
         }
 
         /// <summary>
         /// Asynchronously repeatedly calls the specified polling action, delaying between calls,
         /// until a given condition is met in the response.
         /// </summary>
-        /// <typeparam name="TResponse">The response type. Must not be null.</typeparam>
+        /// <typeparam name="TResponse">The response type.</typeparam>
```
Comment (jskeet) [inline src/Google.Api.Gax/Polling.cs]: Thanks :)

File: src/Google.Api.Gax/PollSettings.cs (hunk 3 above)
Comment (chrisdunelm) [inline src/Google.Api.Gax/PollSettings.cs]: It'd certainly be neat ;) But possibly not very obvious!
This again is copied directly from BackoffSettings.

**Pull request merged**